solarforecastarbiter/validation/validator.py

```diff
@@ -406,3 +407,100 @@ def check_timestamp_spacing(times, freq=None):
         return len(gaps) == 1
     else:
         return True  # singleton DatetimeIndex passes
+
+
+def _zero_runs(a):
+    """ Finds runs of zeros in a.
+
+    Parameters
+    ----------
+    a : array
+
+
+    Returns
+    -------
+    ranges : array
+        Size Nx2, where ranges[k, 0] is the starting position of the kth run,
+        and ranges[k, 1] is the ending position, i.e., a[start:end] is zero.
+
+    Credit to https://stackoverflow.com/a/24892274/8484669
+    """
+    iszero = np.concatenate(([0], np.equal(a, 0).view(np.int8), [0]))
+    absdiff = np.abs(np.diff(iszero))
+    # Runs start and end where absdiff is 1.
+    ranges = np.where(absdiff == 1)[0].reshape(-1, 2)
+    return ranges
+
+
+def detect_stale_values(x, window=3, rtol=1e-5, atol=0):
```

Comment:
I'm concerned about the default `atol` value. From the [isclose doc](https://docs.scipy.org/doc/numpy/reference/generated/numpy.isclose.html): "A zero value for atol will result in False if either a or b is zero." Are nighttime 0s supposed be marked as stale?

Reply:
Agree with the concern.

Hadn't thought about behavior for nighttime zeros.  I'd assume that these data would be marked as stale, since the criteria is that values essentially don't change during a `window`.  Different view?

I should include a sequence of 0's in the tests.

Reply:
I think I'd mark them as stale here. Maybe there is another function later on that combines flags and says something like "problem = not night and stale". I don't think we need to decide on that at the moment, though.

Agree that 0s and near 0s should be tested in either case.